src/SourceBuild/content/eng/tools/BinaryToolKit/BinaryTool/Driver.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Logging;
+
+namespace BinaryTool;
+
+public static class Driver
+{
+    // Directory to scan for binaries: required
+    public static string TargetDirectory { get; set; } = string.Empty;
+
+    // Directory to output reports: required
+    public static string OutputReportDirectory { get; set; } = string.Empty;
+
+    // File for the baseline of allowed retainable binaries: optional
+    public static string AllowedBinariesKeepFile { get; set; } = string.Empty;
+
+    // File for the baseline of allowed removeable binaries: optional
+    public static string AllowedBinariesRemoveFile { get; set; } = string.Empty;
```

Comment:
I'd much prefer these to be parameters to the `Execute` method. This pattern of a whole bunch of static state makes it difficult to understand dependencies and code flow. Static state is also problematic in unit testing scenarios.